scio-avro/src/main/scala/com/spotify/scio/avro/AvroIO.scala

```diff
@@ -131,42 +131,40 @@ object ProtobufIO {
   type WriteParam = AvroIO.WriteParam
   val WriteParam = AvroIO.WriteParam
 }
-
-final case class AvroIO[T: ClassTag: Coder](path: String, schema: Schema = null) extends ScioIO[T] {
-  override type ReadP = Unit
-  override type WriteP = AvroIO.WriteParam
+sealed trait AvroIO[T] extends ScioIO[T] {
   override final val tapT = TapOf[T]
 
-  private def avroOut[U](sc: SCollection[T],
-                         write: beam.AvroIO.Write[U],
-                         path: String,
-                         numShards: Int,
-                         suffix: String,
-                         codec: CodecFactory,
-                         metadata: Map[String, AnyRef]) =
+  protected def avroOut[U](sc: SCollection[T],
+                           write: beam.AvroIO.Write[U],
+                           path: String,
+                           numShards: Int,
+                           suffix: String,
+                           codec: CodecFactory,
+                           metadata: Map[String, AnyRef]) =
     write
       .to(ScioUtil.pathWithShards(path))
       .withNumShards(numShards)
       .withSuffix(suffix)
       .withCodec(codec)
       .withMetadata(metadata.asJava)
 
+}
+
+final case class SpecificRecordIO[T: ClassTag: Coder](path: String)(
+  implicit ev: T <:< SpecificRecordBase)
```

Comment:
@andrisnoko couldn't we have this as `SpecificRecordIO[T <: SpecificRecordBase : ClassTag: Coder]`?